test/udp_sender.js

```diff
@@ -249,4 +250,21 @@ describe('udp sender should', () => {
         assert.equal(response.err, false);
         assert.equal(response.numSpans, 0);
     });
+
+    it ('flush gracefully handles errors emitted by socket.send', done => {
+        sender._host = 'foo.bar.com';
+        sender._port = 1234;
+        new Tracer(
+            'test-service-name',
+            new RemoteReporter(sender),
+            new ConstSampler(true)
+        ).startSpan('testSpan').finish();
+        let oldLog = console.log;
+        sender.flush();
+        console.log = message =>  {
```

Comment:
still a race condition, is there not? The flush might call the log function before you have a chance to replace it.